test/unit/dataContract/stateTransition/validation/validateDataContractSTStructureFactory.spec.js

```diff
@@ -0,0 +1,64 @@
+const validateDataContractSTStructureFactory = require('../../../../../lib/dataContract/stateTransition/validation/validateDataContractSTStructureFactory');
+
+const DataContractStateTransition = require('../../../../../lib/dataContract/stateTransition/DataContractStateTransition');
+
+const getDataContractFixture = require('../../../../../lib/test/fixtures/getDataContractFixture');
+
+const { expectValidationError } = require('../../../../../lib/test/expect/expectError');
+
+const ValidationResult = require('../../../../../lib/validation/ValidationResult');
+
+const ConsensusError = require('../../../../../lib/errors/ConsensusError');
+
+describe('validateDataContractSTStructureFactory', () => {
+  let validateDataContract;
+  let validateDataContractSTStructure;
+  let rawDataContract;
+  let rawStateTransition;
+
+  beforeEach(function beforeEach() {
+    validateDataContract = this.sinonSandbox.stub();
+
+    const dataContract = getDataContractFixture();
+    const stateTransition = new DataContractStateTransition(dataContract);
+
+    rawDataContract = dataContract.toJSON();
+    rawStateTransition = stateTransition.toJSON();
+
+    validateDataContractSTStructure = validateDataContractSTStructureFactory(
+      validateDataContract,
+    );
+  });
+
+  it('should return invalid result if data contract is not invalid', () => {
```

Comment:
Maybe 'should return invalid result if data contract is not invalid'? It looks like the last "not" changes the meaning of that phrase